harper-core/src/document.rs

```diff
@@ -284,6 +284,12 @@ impl Document {
                     }
 
                     let child_tok = &copy[cursor];
+
+                    // Only condense adjacent spans
+                    if start_tok.span.end != child_tok.span.start {
```

Comment:
Add this change to `condense_newlines` as well.

Reply:
I did that originally, but it broke a test so I reverted it.

Reply:
The test might be wrong then.

Reply:
Ah, it's because newlines in comments need to be merged for paragraph breaks.

```
// I am a comment
//
// My span starts after the //
```
There is no paragraph break in between because the newline on line 2's span isn't contiguous with the newline at the end of line 1.